zha/application/discovery.py

```diff
@@ -306,6 +308,27 @@ def discover_quirks_v2_entities(self, device: Device) -> None:
                     )
                     cluster_handler.__dict__[zha_const.ZCL_INIT_ATTRS] = init_attrs
 
+                if (
+                    hasattr(entity_metadata, "attribute_name")
+                    and hasattr(entity_metadata, "reporting_config")
+                    and entity_metadata.reporting_config
+                ):
+                    reporting_config = tuple(
+                        filter(
+                            lambda cfg: cfg["attr"] != entity_metadata.attribute_name,
+                            cluster_handler.REPORT_CONFIG,
+                        )
+                    )
+                    reporting_config += (
+                        AttrReportConfig(
+                            attr=entity_metadata.attribute_name,
+                            config=astuple(entity_metadata.reporting_config),
+                        ),
+                    )
+
+                    cluster_handler.__dict__[zha_const.REPORT_CONFIG] = reporting_config
```

Comment:
The attributes are also added to `ZCL_INIT_ATTRS` above this new code block.

If I'm looking at this correctly, we'll have them in both `REPORT_CONFIG` and `ZCL_INIT_ATTRS` and then have duplicates in `uncached`?
https://github.com/zigpy/zha/blob/11cf9ba5d3a47f8733df0c095befac233acb8b73/zha/zigbee/cluster_handlers/__init__.py#L458-L460

We might want to move the new block before the `ZCL_INIT_ATTRS` block and then check in the `ZCL_INIT_ATTRS` block that they're not already in `REPORT_CONFIG` OR just remove them from `ZCL_INIT_ATTRS` in this line?